opencsp/app/sofast/SofastGUI.py

```diff
@@ -81,15 +62,12 @@ def _get_frame(self) -> np.ndarray:
             2D image from camera.
 
         """
-        # Check camera is loaded
-        if self.image_acquisition is None:
-            messagebox.showerror('Error', 'Camera is not connected.')
-            return
-
-        # Get frame
-        frame = self.image_acquisition.get_frame()
-
-        return frame
+        try:
+            frame = self.service.get_frame()
+            return frame
+        except RuntimeError as ex:
+            messagebox.showerror('Error', repr(ex))
+            return None
```

Comment:
What does the message look like when no camera is connected? I just want to make sure that the message is recognizable to users, (especially simple users who are more likely to be using the GUI).

Reply:
Actually I looked at a preview, and the errors are totally readable, good work. And ideally this would never even get called from the GUI because the buttons are grayed out when components aren't loaded. Feel free to ignore.